app/services/course/assessment/question/programming/cpp/cpp_autograde_pre.cc

```diff
@@ -0,0 +1,132 @@
+// Variant type that encompasses all the types that expect_equal() supports
```

Comment:
Could you also just add in the 'why' for this part, on top of the 'what'? At least this explains the issue, and how the overloading here overcomes that. Might be helpful for someone who wishes to 'refactor' this in the future, or extend it.